Remove test timings

rcomer opened the pull request:
## 🚀 Pull Request

### Description
<!-- Provide a clear description about your awesome pull request -->
<!-- Tell us all about your new feature, improvement, or bug fix -->
Here is some more Iris functionality that I did not know existed, until @pp-mo mentioned it the other day.  Described by the removed comment
https://github.com/SciTools/iris/blob/6eb0401011a8edce51a942a176e2feb06f4b3f09/lib/iris/tests/__init__.py#L940-L946

I tried to run this with `unittest discover` as the comment recommends, but got errors, particularly from [test_pandas.py](https://github.com/SciTools/iris/blob/main/lib/iris/tests/unit/pandas/test_pandas.py).  I found that this functionality can be used with `pytest -s -v`, so if we do want to keep the functionality we should probably document that.  I'm unsure if it can be made to work with `pytest-xdist`.

In any case, pytest provides its own functionality to tell you the timings of the slowest running tests via the `--durations` flag, and of course we now have some `pytest` tests which don't inherit from `IrisTest`, so won't work with the bespoke functionality unless we go through and add the decorator.

More generally, the more we can simplify these test classes, the better chance we have of eventually ditching `unittest` altogether.

This functionality was added at #2372.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

lbdreyer reviewed (changes requested):
Looking great! Thanks for doing this @rcomer !

I spotted one other mention of IrisTest_nometa here: https://github.com/SciTools/iris/blame/main/lib/iris/tests/graphics/README.md#L27

Do you mind also updating that in this PR?

rcomer commented:
Thanks @lbdreyer, good spot!  I only grepped the python files.  Now fixed and also rebased to address the whatsnew conflict.  I don't know why the CI isn't running 😕
